detail.py

```diff
@@ -0,0 +1,29 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from datetime import datetime, timezone
+from catalog import models as cmod
+
+
+@view_function
+def process_request(request, prod:cmod.Product=None):
+
+    product = cmod.Product.objects.get(id=prod.id)
```

Comment:
I think that this code is redundant. You could just use the variable prod instead